### PR TITLE
Integration Testing Pipe

### PR DESCRIPTION
**Pull request by gowrizrh**
I've committed to the main branch in order to have the image built and tested as a starting point. Now that I'm done this PR is into a dummy branch purely only for code review.

This likely is my first bash scripting attempt. Feel free to suggest even the slightest improvement :memo:

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FROM php:8.1-alpine
```
Comment (gowrizrh) [inline Dockerfile:1]: note: please expand and review this entire Dockerfile

File: README.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+| `ELASTICSEARCH_HOST`    | Must be `host.docker.internal` when running in pipelines. Optionally change this if you are developing the pipe locally.                                   |
+| `RABBITMQ_HOST`         | Must be `host.docker.internal` when running in pipelines. Optionally change this if you are developing the pipe locally.                                   |
+| `DATABASE_HOST`         | Must be `host.docker.internal` when running in pipelines. Optionally change this if you are developing the pipe locally.                                   |
+| `COMPOSER_AUTH`         | JSON stringified composer `auth.json` with the relevant configuration you need.                                                                            |
+| `REPOSITORY_URL`        | `https://repo.magento.com/` - If using this, make sure the `COMPOSER_AUTH` variable is set. <br>  `https://mirror.mage-os.org/` - Only supports open source edition. <br> Default: `https://repo.magento.com/` |
+| `MAGENTO_VERSION`       | (Optional) Default: `magento/project-community-edition:>=2.4.5 <2.4.6`                                                                                     |
```
Comment (aligent-lturner) [inline README.md]: Can give an example for commerce here too?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+| `GROUP`                 | (Optional) Specify a test group to run. Example: `--group inventory indexer_dimension`                                                                     |
```
Comment (aligent-lturner) [inline README.md]: Is this two groups or one? Not completely clear from an instructional point-of-view. Could potentially also reference the `@group` annotation

File: README.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+| `TESTS_PATH`            | (Optional) Specify a test path to run. Example `./app/code/The/Module`                                                                                     |
+
+## Example Pipeline
```
Comment (aligent-lturner) [inline README.md:31]: Can one of the examples below use the `GROUP` variable?

File: README.md (hunk 2 above)
Comment (gowrizrh) [reply README.md]: I've added this

File: README.md (hunk 3 above)
Comment (gowrizrh) [reply README.md]: This should be updated now

File: README.md (hunk 4 above)
Comment (gowrizrh) [reply README.md:31]: Added an example for `GROUP` var usage

**Comment by aligent-lturner** [comment]
@gowrizrh Looks ok to me, but I'd wait for a review from devops.
Might be worth adding a CODEOWNERS file and requiring owner approval for PRs

File: pipe.sh [hunk 5]
```diff
@@ -1,19 +1,160 @@
...
+    composer create-project --repository-url="$REPOSITORY_URL" "$MAGENTO_VERSION" ./magento2 --no-install
+    cd magento2
+  fi
+
+  composer config --no-interaction allow-plugins.dealerdirect/phpcodesniffer-composer-installer true
+  composer config --no-interaction allow-plugins.laminas/laminas-dependency-plugin true
+  composer config --no-interaction allow-plugins.magento/* true
+  cat composer.json
+  composer install
+  cd dev/tests/integration
```
Comment (daniel-vdp) [inline pipe.sh:51]: These 6 or so lines are duplicated a few times. Can we pull this into a `composer_config_setup()` function or something similar?

**Review by daniel-vdp — CHANGES_REQUESTED** [review]
Minor comment but looks good to me :tada:

File: pipe.sh (hunk 5 above)
Comment (aligent-lturner) [reply pipe.sh:51]: Yep - the latest commit showed the need for this. Extra lines were added only to the graphql function and not the other 2

Comment (gowrizrh) [reply pipe.sh:51]: thanks, i'll resolve this next

Comment (gowrizrh) [reply pipe.sh:51]: this is resolved by #2

**Review by daniel-vdp — APPROVED** [review]
LGTM

**Pull request merged**